### PR TITLE
Added possibility to change step limit via frontend

### PR DESCRIPTION
**Pull request by Mr-Groch**
## Description

Added to global config 'STEP_LIMIT'. Initial value readed from get_args(). Used in search_loop() instead of args.step_limit and possible to change via web nav panel. Changes are instant, because this adds also dummy 'NEXT_LOCATION' (actual scan location).

EDIT:
Also added -nst (--no-step-control) arg to disable step limit change from website

## Motivation and Context

Dynamic step limit is usefull when I travel from country side to big city. In big city big step_limit is useless, because i can't travel fast in straight direction, I just wanna quick search results.

## How Has This Been Tested?

Changing step limit without changing location, with lcoation change, after location change and before location change.

## Screenshots (if appropriate):
steplimit

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by DeviantEng** [comment]
I can see the benefit of this is some environments, but I do not want this at all in mine.  In fact, I would like to remove the `Search` toggle from the options area.  Maybe a setting in the config.ini to disable certain toggles?

**Comment by aaitmouloud** [comment]
@DeviantEng I don't think that's the main subject here, just create an issue or a PR about what you are asking for

**Comment by Mr-Groch** [comment]
Added one more change after PR #2956 - current_location for Flask were not updated, so after page refresh, marker always have been returning to origin postion.

**Comment by mackhankins** [comment]
@DeviantEng -nsc for search toggle.

File: pogom/app.py [hunk 1]
```diff
@@ -142,6 +142,7 @@ def next_loc(self):
             return 'bad parameters', 400
         else:
             self.location_queue.put((lat, lon, 0))
+            self.current_location = (lat, lon)
```
Comment (chuyskywalker) [inline pogom/app.py]: Ah, thanks, I'll grab this one.

Comment (chuyskywalker) [inline pogom/app.py]: Or I won't since I didn't notice that I'd already been merged :D

**Comment by giejay** [comment]
This is awesome, when is this gonna get merged?

**Comment by AHAAAAAAA** [comment]
+1

**Comment by Mr-Groch** [comment]
@DeviantEng added -nst (--no-step-control) arg to disable this from website in commit 592a386015a82848d1cb4ce4f4fc0410d2b87e17

**Comment by Mr-Groch** [comment]
Merged with upstream and updated screenshot

**Comment by Mr-Groch** [comment]
Why closed?

**Comment by cris11368** [comment]
Glad we didn't loose all the requests and changes can still be done manually....

**Pull request closed without merging**